lib/zipkin/scope.rb

```diff
@@ -0,0 +1,36 @@
+module Zipkin
+  # Scope represents an OpenTracing Scope
+  #
+  # See http://www.opentracing.io for more information.
+  class Scope
+    def initialize(span, scope_stack, finish_on_close:)
+      @span = span
+      @scope_stack = scope_stack
+      @finish_on_close = finish_on_close
+      @closed = false
+    end
+
+    # Return the Span scoped by this Scope
+    #
+    # @return [Span]
+    attr_reader :span
+
+    # Close scope
+    #
+    # Mark the end of the active period for the current thread and Scope,
+    # updating the ScopeManager#active in the process.
+    def close
+      raise "Tried to close already closed span: #{inspect}" if @closed
```

Comment:
I'm not entirely familiar with this project and it's conventions, but typically I do not want errors in instrumentation or tracing code to crash an application. For that reason, I would log a warning, but not raise an exception. However, I realize that opinions vary in this regard.

Reply:
I mostly agree however this seems very exceptional case. I will do some testing and will change this if it turns out to be a problem.